### PR TITLE
[Data Plane Mgr] UT Test Automation - Part II

### PR DESCRIPTION
**Pull request by kevin-zhonghao**
The second version of DPM Auto test. 
It can provide custom parameter options, support more flexible test scenarios

### CODE REVIEW

**Comment by xieus** [comment]
@kevin-zhonghao Let us go through this PR together today.

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/UTL3NeighborInfoMapping.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.dataplane.entity;
+
+import java.util.List;
+
+public class UTL3NeighborInfoMapping {
```
Comment (xieus) [inline services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/UTL3NeighborInfoMapping.java:20]: Is this possible to move three new UT classes to under /test as they are pure test classes?

Comment (kevin-zhonghao) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/UTL3NeighborInfoMapping.java:20]: Sure

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/entity/UTIPInfo.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.dataplane.utils.entity;
+
+public class UTIPInfo {
```
[the remark this answers is not in the document]
Comment (kevin-zhonghao) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/entity/UTIPInfo.java:18]: Sure

File: web/src/main/java/com/futurewei/alcor/web/entity/dataplane/InternalPortEntity.java [hunk 3]
```diff
@@ -28,6 +29,7 @@
     private List<RouteEntity> routes;
 
     @JsonProperty("binding_host_ip")
+    @SerializedName("binding_host_ip")
```
[the remark this answers is not in the document]
Comment (kevin-zhonghao) [reply web/src/main/java/com/futurewei/alcor/web/entity/dataplane/InternalPortEntity.java:32]: This annotation is for maintaining the Json name when we use Gson.toJson() method

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/entity/UTL3NeighborInfoMapping.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        limitations under the License.
+*/
+package com.futurewei.alcor.dataplane.utils.entity;
+
+import java.util.List;
+
+/**
+ * Used for configuring NeighborTable, subnet and its IPs mapping
+ */
+public class UTL3NeighborInfoMapping {
```
[the remark this answers is not in the document]
Comment (kevin-zhonghao) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/entity/UTL3NeighborInfoMapping.java:23]: Thanks for the link. The current UT name follows the rule.

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/entity/UTIPInfo.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (kevin-zhonghao) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/entity/UTIPInfo.java]: These entities are related to auto UT util class, so they may have to be in entity package

**Review by xieus — APPROVED** [review]

**Pull request merged**